src/lxc/lxccontainer.c

```diff
@@ -2765,29 +2765,47 @@ static bool do_lxcapi_destroy_with_snapshots(struct lxc_container *c)
 
 WRAP_API(bool, lxcapi_destroy_with_snapshots)
 
-static bool set_config_item_locked(struct lxc_container *c, const char *key, const char *v)
+int lxc_set_config_item_locked(struct lxc_conf *conf, const char *key,
+			       const char *v)
 {
+	int ret;
 	struct lxc_config_t *config;
+	bool bret = true;
+
+	config = lxc_get_config(key);
+	if (!config)
+		return -EINVAL;
+
+	ret = config->set(key, v, conf, NULL);
+	if (ret < 0)
+		return -EINVAL;
+
+	if (lxc_config_value_empty(v))
+		do_clear_unexp_config_line(conf, key);
+	else
+		bret = do_append_unexp_config_line(conf, key, v);
+	if (!bret)
+		return -ENOMEM;
+
+	return 0;
+}
+
+static bool do_set_config_item_locked(struct lxc_container *c, const char *key,
```

Comment:
Generally fn() would call do_fn(), right?  This do_fn() and lxc_fn() is confusing and might become problematic...  Unfortunately I don't have any better ideas for naming.  Do you?  (If not, then I guess we leave it as is)

Reply:
Yeah, I mean we can reverse the naming for the two functions and export the `do_fn()` method in `lxc.h`. Wouldn't matter much I'd say.

Reply:
Yeah, let's leave it.  Thanks.